File: src/3074.apple-redistribution-into-boxes.3334/solution.ts
function minimumBoxes (apple: number[], capacity: number[]): number {
    let sum = 0;
    for (const n of apple) {
        sum += n;
    }
    let i = 0;
    capacity.sort((a, b) => b - a);
    for (;i < capacity.length; i++) {
        if (sum <= 0) {
            break;
        }
        sum -= capacity[i];
    }
    return i;
}

File: src/3074.apple-redistribution-into-boxes.3334/solution.java
class Solution {
    public int minimumBoxes(int[] apple, int[] capacity) {
        int sum = 0;
        for(int a : apple){
            sum += a;
        }
        Arrays.sort(capacity);
        int index = capacity.length-1;
        while (index>-1 && sum>0){
            sum -= capacity[index--];
        }

        return capacity.length-index-1;
    }
}